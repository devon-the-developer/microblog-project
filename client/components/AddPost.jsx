import React from 'react'

export class AddPost extends React.Component {
    constructor(props) {
        super(props) 
        this.state = {
            postTitle: null,
            postTags: null,
            postContent: null
        }
        this.handleChange = this.handleChange.bind(this)
    }

    handleChange(event) {
        const {name, value} = event.target
        this.setState({
            [name]:value
        })
        console.log(event.target.name)
    }

    render() {
        return (
            <div>
                <form>
                    <label>Title: </label>
                    <input type='text' name='postTitle' onChange={this.handleChange}></input>
                    <br />
                    <label>Tags: </label>
                    <input type='text' name='postTags' onChange={this.handleChange}></input>
                    <br />
                    <label>Content: </label>
                    <input type='text' name='postContent' height='200px' onChange={this.handleChange}></input>
                    <br />
                    <button>Submit</button>
                </form>
                <p>{this.state.postTitle} {this.state.postContent}</p>
            </div>
        )
    }
}