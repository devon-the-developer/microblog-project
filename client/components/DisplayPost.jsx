import React from 'react'

export default class DisplayPost extends React.Component {
    constructor(props) {
        super(props)

        this.state= {
            posts: '',
            currentId: ''
        }
    }

    componentWillMount(){
        console.log('WillMount')
        this.renderPost()
    }

    renderPost = () => {
        console.log('renderPost')
        this.setState({
            posts: this.props.value,
            currentId: parseInt(this.props.match.params.id) 
        })
    }

    render(){
       
        // const currentId = parseInt(this.props.match.params.id) + 1
        console.log("currentId + 1: ", this.state.currentId) 
        // const currentPost = this.props.value.find(post => post.id == currentId) 
        // console.log("currentPost: ", currentPost)
        console.log('this.state.posts :', this.state.posts)
        console.log("currentPost via state: ", this.state.posts.find(post => post.id == this.state.currentId))
        const currentPost = this.state.posts.find(post => post.id == this.state.currentId) || ''
        return(
            <div>
            <h2>{currentPost.name}</h2>
            <span><strong>Tags: </strong>{currentPost.tags}</span>
            <p>{currentPost.content}</p>


            </div>
        )
    }
}