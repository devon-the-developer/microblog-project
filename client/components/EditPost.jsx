import React from 'react'
import { connect } from 'react-redux'

import { editPost } from '../api'
import { fetchBlogPosts } from '../actions';

export class EditPost extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            nameChangeTo: null,
            tagsChangeTo: null,
            contentChangeTo: null
        }
    }

    componentDidMount() {
        const currentPost = this.props.data.posts.find(post => post.id == this.props.data.currentPostId) || ''
        this.setState({
            nameChangeTo: currentPost.name,
            tagsChangeTo: currentPost.tags,
            contentChangeTo: currentPost.content
        })
    }

    handleChange = (event) => {
        this.setState({
                [event.target.name]: event.target.value
        })
    }

    handleSubmit = (event) => {
        event.preventDefault()
        const editObject = {
            id: this.props.data.currentPostId,
            name: this.state.nameChangeTo,
            tags: this.state.tagsChangeTo,
            content: this.state.contentChangeTo
        }
        editPost(editObject)
        .then(this.props.getPostsAgain())
        this.props.history.push('/')
    }

    render(){
        const currentPost = this.props.data.posts.find(post => post.id == this.props.data.currentPostId) || ''
        return (
            <div className='blogpost'>
                <form>
                    <h4>Edit Post</h4>
                    <label>Title: </label>
                    <input type='text' name='nameChangeTo' onChange={this.handleChange} defaultValue={currentPost.name}></input>
                    <br />
                    <label>Tags: </label>
                    <input type='text' name='tagsChangeTo' onChange={this.handleChange} defaultValue={currentPost.tags}></input>
                    <br />
                    <label>Content: </label>
                    <textarea type='text' name='contentChangeTo' onChange={this.handleChange} defaultValue={currentPost.content} rows='50' cols='100'></textarea>
                    <br />
                    <button onClick={this.handleSubmit}>Save Changes</button>
                </form>
            </div>
        )
    }

}

const mapStateToProps = (state) => {
    return {
        data: state.blogposts
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
       getPostsAgain: () => dispatch(fetchBlogPosts())
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditPost)