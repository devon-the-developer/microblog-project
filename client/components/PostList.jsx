import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'

class PostList extends React.Component {
    render() {
        return (
            <div>
                {this.props.data.posts.map((post, index) => 
                    <p key={index}>
                        <strong>
                            <Link to={`/post/${post.id}`}>
                                {post.name}
                            </Link>
                        </strong>
                        <br />
                        {post.content}
                    </p>
                    )
                }
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        data: state.blogposts
    }
}
export default connect(mapStateToProps)(PostList)