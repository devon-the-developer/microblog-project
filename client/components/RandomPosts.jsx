import React, {Fragment} from 'react'
import { Link } from 'react-router-dom'

import { connect } from 'react-redux'

export class RandomPosts extends React.Component {

    render(){
        const amountOfPosts = this.props.data.posts.length
        const totalPostsToShow = 5
        const randomPostsToShow = this.props.data.posts.filter(post => post.id === (Math.floor((Math.random() * amountOfPosts))) )
        console.log(randomPostsToShow)
        return (
            <div>
                <h4>Random Posts</h4>
                {randomPostsToShow.map((post, index) => <p key={index}><strong><Link to={`/post/${post.id}`}>{post.name}</Link></strong><br />{post.content}</p>)}
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        data: state.blogposts || []
    }
}

export default connect(mapStateToProps)(RandomPosts)