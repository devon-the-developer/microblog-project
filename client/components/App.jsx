import React from 'react'
import {HashRouter as Router, Route, Link} from 'react-router-dom'

import { AddPost } from './AddPost'
import  RandomPosts  from './RandomPosts'
import { EditPost } from './EditPost'
import { getPosts } from '../api'
import DisplayPost from './DisplayPost'
import { connect } from 'react-redux';
import { fetchBlogPosts } from '../actions';


class App extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      error: null,
      posts: []
    }
  }

  componentDidMount() {
    this.props.getPosts()
  }

  render() {
    const basicStyle = {
      padding: 50,
      background: '#abbaab',
      background: '-webkit-linear-gradient(to top, #ffffff, #abbaab)',/* Chrome 10-25, Safari 5.1-6 */
      background: 'linear-gradient(to top, #ffffff, #abbaab)' /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */

    }
    return (
      <Router>
        <div style={basicStyle}>
          <h1><strong>MicroBlog Project</strong></h1>
          <div style={{display:'inline-block'}}>
            <Link to='/'>Home</Link> <Link to='/addpost'>Add Post</Link>
          </div>
          <br />
          <br />
          <div>
            <Route exact path='/' component={RandomPosts} />
            <Route exact path='/post/:id' render={(props) => <DisplayPost {...props} value={this.state.posts} />} />
            <Route path='/addpost' component={AddPost} />
            <Route exact path='/editpost/:id' render={(props) => <EditPost {...props} value={this.state.posts} />} />
          </div>
        </div>
      </Router>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    data: state.blogposts
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    getPosts: () => dispatch(fetchBlogPosts())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(App)

