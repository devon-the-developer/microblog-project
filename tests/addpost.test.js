import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { AddPost } from '../client/components/AddPost'

describe('<AddPost />', () => {
    it('has 2 <input /> elements', () => {
        const expected = 2
        const wrapper = shallow(<AddPost />)

        const actual = wrapper.find('input')

        expect(actual.length).toBe(expected)
    })
    it('has 1 <textarea /> element', () => {
        const expected = 1 
        const wrapper = shallow(<AddPost />)

        const actual = wrapper.find('textarea')

        expect(actual.length).toBe(expected)
    })
    it('should match snapshot', () => {
        const tree = shallow(<AddPost />)

        expect(toJson(tree)).toMatchSnapshot()
    })
})
