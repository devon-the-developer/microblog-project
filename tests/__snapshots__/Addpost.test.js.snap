// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddPost /> should match snapshot 1`] = `
<div>
  <form>
    <h4>
      Add Post
    </h4>
    <label>
      Title: 
    </label>
    <input
      name="name"
      onChange={[Function]}
      type="text"
    />
    <br />
    <label>
      Tags: 
    </label>
    <input
      name="tags"
      onChange={[Function]}
      type="text"
    />
    <br />
    <label>
      Content: 
    </label>
    <textarea
      cols="100"
      name="content"
      onChange={[Function]}
      rows="50"
      type="text"
    />
    <br />
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </form>
</div>
`;
